### ISSUE TITLE
Make IDToken extendable

### ISSUE THREAD

**Issue by CapoD** [HarryMWinters/fastapi-oidc#14]
Hi there,

thanks for sharing your project. We are currently implementing id_token validation in our project and found your library quite useful. 

We would like to discuss a thought with you: 
How about making the IDToken class configurable such that one may provide a Subclass with additional fields and validations? 
Needless to say that IDToken currently allows for additional fields, but we believe that our idea might make some things easier.

**Comment by HarryMWinters** [comment]
Hey @CapoD,

I'm glad to hear this repo has been of use to someone and I appreciate your engagement.

I think your idea about subclassing the IDToken is a good one. It would make this library much more usable. What do you think of something like this?

```python3
from fastapi_oidc import IDToken

class CustomIDToken(IDToken):
    """
    This would be a custom IDToken with which has pydantic methods as well as the required fields from IDToken.
    """
    custom_field: str
    other_custon_field: int


OIDC_config = {
    "client_id": "0oa1e3pv9opbyq2Gm4x7",
    "base_authorization_server_uri": "https://dev-126594.okta.com",
    "issuer": "dev-126594.okta.com",
    "signature_cache_ttl": 3600,
}

# The custom_id_token kwarg would be optional and authenticate_user 
# would return a base IDToken by default.
authenticate_user: Callable = get_auth(**OIDC_config, custom_ID_token=CustomIDToken)

```

**Comment by CapoD** [comment]
@HarryMWinters that is exactly what we need 👍

**Comment by HarryMWinters** [comment]
:) Sweet. I'll implement it in the next week or two.

**Comment by HarryMWinters** [comment]
See PR #18. This is merged and released in `v0.0.7`. Good luck @CapoD and thanks for the feedback.
